cmd/fleetctl/fleetctl/gitops.go

```diff
@@ -60,6 +60,19 @@ func gitopsCommand() *cli.Command {
 			debugFlag(),
 		},
 		Action: func(c *cli.Context) error {
+			logf := func(format string, a ...interface{}) {
+				_, _ = fmt.Fprintf(c.App.Writer, format, a...)
+			}
+
+			// Make sure to capture the --dry-run flag in case it's passed as a positional argument
+			// which can happen when using glob patterns (e.g. -f my_dir/* --dry-run).
+			// We only want to do this for --dry-run due to its importance in preventing unwanted side effects.
+			for _, arg := range c.Args().Slice() {
+				if arg == "--dry-run" {
+					flDryRun = true
+				}
+			}
```

Comment:
Can we just bail if we see positional arguments at all? `fleetctl gitops` doesn't actually have any positional arguments, so if we do a glob like `-f my_dir/*` it should fail anyway since you're not gonna get what you expect.